app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt

```diff
@@ -165,6 +167,10 @@ class AddLoginFragment : Fragment(R.layout.fragment_add_login), MenuProvider {
                             setHostnameError()
                             binding.clearHostnameTextButton.isEnabled = true
                         }
+                        hostnameContainsProtocol && !hostnameText.startsWith("http") -> {
```

Comment:
Because above this code there is a check for `http|https|rtsp|ftp` it seems like this we would show the `add_login_hostname_invalid_text_3` error just for the `rtsp|ftp` protocols.
This might be okay - we would still be showing an error for other protocols but it shows that we have duplicated code for checking the protocol so the new case introduced here should be merged with the already existing `!Patterns.WEB_URL.matcher(hostnameText).matches()` check. Maybe we can replace the existing one?

Reply:
It could, though I think having a specific error message for non-https protocols is more user-friendly. Are you suggesting to combine the `!hostnameText.startsWith("http")` check with the previous check?

In any case, `hostnameContainsProtocol` is useful for the auto-`https` code, so the end result would be quite similar (`!hostnameText.startsWith("http")` would be converted to a `val` to be both checked and passed to `setHostnameError`).

Reply:
> the end result would be quite similar (!hostnameText.startsWith("http") would be converted to a val to be both checked and passed to setHostnameError). 

Sounds good.
Otherwise it seems to me that the above check for `!Patterns.WEB_URL.matcher(hostnameText).matches()` would catch most of the other protocols and instead of the intent here to show the `Web address must contain "https://" or "http://"` error we would show `Valid hostname required`.

Reply:
> Otherwise it seems to me that the above check for !Patterns.WEB_URL.matcher(hostnameText).matches() would catch most of the other protocols and instead of the intent here to show the Web address must contain "https://" or "http://" error we would show Valid hostname required.

I'm a bit confused here, in what cases would it show `Valid hostname required` where you say it should show `Web address must contain "https://" or "http://"`? My intent is to only show the `http(s)` error string if `hostnameText` is a valid non-http(s) URL, so an invalid `hostnameText` URL should indeed show `Valid hostname required` before the `http(s)` error string.

It turns out that merging the two branches with no logic change results in messier code IMO, so I think if no logic change is required then it would be better to leave this part alone.

Reply:
There are two checks for protocols. [One previously existing](https://github.com/mozilla-mobile/fenix/blob/48288f0198394b4e57d14549682f4d22a04ea30e/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt#L164-L167) and the new one you are adding just below.
This would mean that a valid url using a different protocol than what `Patterns.WEB_URL` matches would show `Valid hostname required` and only urls with protocols matching `Patterns.WEB_URL` but not using `https` / `https` would show `Web address must contain "https://" or "http://"`.

https://user-images.githubusercontent.com/11428869/202380006-de6ecfe2-4f43-4131-b72e-b4bc90e1e8b8.mov


To me it looks like this checks should be combined (by removing the current `Patterns.WEB_URL` match):
- if there is a protocol we should match it with your new `http(s)` check or show the `Web address must contain "https://" or "http://"` error.
- if there is no protocol and the url is invalid then show the `Valid hostname required` error.